### ISSUE TITLE
[BUG] Use SPACE to restart the finished video

### ISSUE THREAD

**Issue by LangLangBart** [iina/iina#4562]
**System and IINA version:**

- macOS 10.15
- IINA 1.3.3

**Expected behavior:**
- See the video below, the video has reached the end and I just pressed <kbd>Space</kbd> to start from the beginning.
  - the video was created with version `1.3.2` 

https://github.com/iina/iina/assets/92653266/8ebe15ab-c3dd-4faa-9ac7-be3769c0cb34

- you may have to enable `Resume last playback position`, if not already
![SCR-20230806-ohiy](https://github.com/iina/iina/assets/92653266/b06c835d-d136-4537-b5d6-ebe31953a0f8)

**Actual behavior:**
- Pressing <kbd>Space</kbd> on a video that has reached the end does nothing in version `1.3.3`.

**Steps to reproduce:**
- open `IINA`
- start a video and wait until the video is over
- press <kbd>Space</kbd>

**Bisect**
- checking [Nightly builds](https://nightly.iina.io/)
- the issue can first be noticed with 0bc3691
  - kindly pinging @svobs  

```sh
2023-07-15 01:58:10 | 0bc3691 | Matt Svoboda

git log -1 0bc3691
# commit 0bc3691cf5805f51ddb491ee249c4d75e056d147
# Author: Matt Svoboda <matt.svoboda@gmail.com>
# Date:   14/Jul/23
 
# Fix TAB & ESC key bindings not working in main player & music mode windows.
# Remove previous workarounds for  ESC & SPACE since they are no longer necessary. (#4464)
```

**Comment by low-batt** [comment]
Thank you for the excellent detailed report.

Reproduced. Confirmed IINA 1.3.2 working correctly. This is a regression added in 1.3.3. I also confirmed the fix for issue #4538 does not fix this issue.

**Comment by svobs** [comment]
This looks like a very special case which intersects issue #3785.

I confirmed through debugging that the `cycle pause` command is sent to mpv directly. But because this bypasses the menu item bindings, it doesn't call `PlayerCore.resume()`, which adds the IINA-specific functionality of restarting playback at the end. However...

While testing, I then stumbled upon yet another issue, which is that the above behavior was not consistent even in IINA 1.3.2, if more than one key was bound to the `cycle pause` action. Because only one key mapping can be bound to a menu item at a time, and because we have made this the first valid mapping found, only the first mapping will call `PlayerCore.resume()`. 

So, adding a duplicate mapping like this turns out to be a viable workaround for #3785:
<img width="391" alt="SCR-20230806-qwfg" src="https://github.com/iina/iina/assets/2213815/13916fdc-a0db-4487-b2bd-80511cf76f9a">

But I would argue that this is yet another bug. Even if only the first matched key mapping is displayed in the menu item, all the mappings for the same action should behave identically.

I need to think about it a bit more but I may be able to solve both this issue and the newly discovered issue. We have a one-way association (sort of) from the menu item to the action via the menu item's `representedObject` (see `MenuController.updateKeyEquivalentsFrom()`), but we need to associate the other direction too. Could maybe accomplish this by storing a reference to the menu item in the`KeyMapping` itself while binding the menu items. Then when the binding is matched, the menu item's action selector could be called instead of the raw mpv string. Will play around with this...

**Comment by LangLangBart** [comment]
This issue appears to have been resolved in iina version `1.3.4`, with the fix included in commit: 06be87aa.


```bash
git log -1 06be87aa
#   commit 06be87aafb4bf3099ee4354799d4e1032a06c759
#   Author: Collider LI <lhc199652@gmail.com>
#   Date:   14/Aug/23

#       Add the `input` module for the plugin system (#4569)

#       * Plugin: Add input API

#       * Add getAllKeyBindings API, provide default priority values

#       * Fix some menu items not working after commit 0bc3691 (#4464)

#       * Add back the ability to handle mouseDown and mouseDrag
```

**Comment by low-batt** [comment]
@LangLangBart Thank you very much for pointing this out. Very helpful.

Closing as fixed in IINA 1.3.4.
